pkg/gen/node-templates/provider.ts.mustache

```diff
@@ -166,12 +166,13 @@ export namespace yaml {
 
         const kind = obj["kind"];
         const apiVersion = obj["apiVersion"];
-        if (kind == null || apiVersion == null) {
+        const meta = obj["metadata"];
+        if (kind == null || apiVersion == null || meta == null) {
             return null;
         }
 
-        let id: string = obj["metadata"]["name"];
-        const namespace = obj["metadata"]["namespace"] || undefined;
+        let id: string = meta["name"];
```

Comment:
There's no guarantee that this is a string, right?

Reply:
There is if it's a Kubernetes resource.

Reply:
OK - not sure if we care about arbitrary input here, but I'm fine with it if you're fine.